src/os_image_nodelet.cpp

```diff
@@ -55,9 +56,29 @@ class OusterImage : public nodelet::Nodelet {
     }
 
     std::string get_metadata(ros::NodeHandle& nh) {
+        std::string config;
+        // Setup subscriber to get config via topic
+        auto config_handler = [&](const std_msgs::String::ConstPtr& p) {
+            config = p->data;
+        };
+        auto config_sub = nh.subscribe<std_msgs::String>("config", 1, config_handler);
+  
+        // Setup client to get config via service
         ouster_ros::GetMetadata request;
         auto client = nh.serviceClient<ouster_ros::GetMetadata>("get_metadata");
-        client.waitForExistence();
+  
+        // Wait for a valid config (via topic) or a valid service client
+        while(config.empty() && !client.exists()){
+            NODELET_WARN_THROTTLE(10.0, "OusterImage: waiting for config...");
+            ros::spinOnce();
+            ros::Duration(0.1).sleep();
+        }
```

Comment:
If we get right of the client then we need to add while a check to make sure ros::ok or wasn't not shutdown while we are waiting for the metadata to come in.